### PR TITLE
feat: added command to start ekco storage migration

### PR DESCRIPTION
**Pull request by ricardomaraschini**
#### What this PR does / why we need it:

This PR creates a new command on `kurl` binary. This new command starts up the storage migration through `ekco` operator. The following are the supported flags:
```
$ kurl cluster migrate-multinode-storage --help
Migrate persistent volumes from 'scaling' to 'distributed' storage classes.

Usage:
  kurl cluster migrate-multinode-storage [flags]

Flags:
  -h, --help                       help for migrate-to-distributed-storage-class
      --migrate-timeout duration   Timeout waiting for the storage migration to finish. (default 8h0m0s)
      --ready-timeout duration     Timeout waiting cluster to be ready for the storage migration. (default 10m0s)

Global Flags:
      --debug   enable debug logging
$
```
This is an example of the command's output:
```
$ kurl cluster migrate-multinode-storage
2023/05/19 12:05:56 reading ekco operator credentials.
2023/05/19 12:05:56 finding ekco operator http address.
2023/05/19 12:05:57 waiting cluster to report as ready for storage migration.
2023/05/19 12:05:57 cluster reporting ready for migration: true
2023/05/19 12:05:57 approving cluster storage migration.
2023/05/19 12:05:57 migration has been successfully approved, waiting for it to finish (8h0m0s timeout)...
2023/05/19 12:06:02 cluster reported migration status: "running pvc migration"
2023/05/19 12:06:07 cluster reported migration status: "running pvc migration"
2023/05/19 12:06:12 cluster reported migration status: "running pvc migration"
2023/05/19 12:06:17 cluster reported migration status: "running pvc migration"
2023/05/19 12:06:22 cluster reported migration status: "completed"
$
```

### CODE REVIEW

**Review by camilamacedo86 — DISMISSED** [review]
It shows very good 🥇

**Review by rrpolanco — DISMISSED** [review]
lgtm. I just have some clarification questions.

File: pkg/cli/cluster_migrate_multinode_storage.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+	}
+	cmd.Flags().DurationVar(&opts.readyTimeout, "ready-timeout", 10*time.Minute, "Timeout waiting cluster to be ready for the storage migration.")
+	cmd.Flags().DurationVar(&opts.migrateTimeout, "migrate-timeout", 8*time.Hour, "Timeout waiting for the storage migration to finish.")
+	return cmd
+}
+
+// getEkcoStorageMigrationAuthToken parses the ekco config map and looks for the storage_migration_auth_token property.
+// this field may be empty, on this case this function returns an empty string.
+func getEkcoStorageMigrationAuthToken(ctx context.Context, cli kubernetes.Interface) (string, error) {
+	cm, err := cli.CoreV1().ConfigMaps(ekcoNamespace).Get(ctx, ekcoConfigConfigMap, metav1.GetOptions{})
```
[the remark this answers is not in the document]
Comment (ricardomaraschini) [reply pkg/cli/cluster_migrate_multinode_storage.go:70]: See my comment [here](https://app.shortcut.com/replicated/story/76836/prompt-to-migrate-from-openebs-to-rook#activity-77137). It is not clear to me what was planned on this regard.

[the remark this answers is not in the document]
Comment (emosbaugh) [reply pkg/cli/cluster_migrate_multinode_storage.go:70]: I'm going to attempt to answer this question and I think the reason it is not in a secret in the cluster is that we need to be able to generate a migration from a worker node that technically does not have kubectl access.

File: pkg/cli/cluster_migrate_multinode_storage.go [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+				return fmt.Errorf("failed to get ekco storage migration auth token: %w", err)
+			}
+			opts.log.Printf("finding ekco operator http address.")
+			if opts.ekcoAddress, err = getEkcoAddress(cmd.Context(), client); err != nil {
+				return fmt.Errorf("failed to get ekco address: %w", err)
+			}
+			return runStorageMigration(cmd.Context(), opts)
+		},
+	}
+	cmd.Flags().DurationVar(&opts.readyTimeout, "ready-timeout", 10*time.Minute, "Timeout waiting cluster to be ready for the storage migration.")
```
Comment (emosbaugh) [inline pkg/cli/cluster_migrate_multinode_storage.go:62]:
```suggestion
	cmd.Flags().DurationVar(&opts.readyTimeout, "ready-timeout", 10*time.Minute, "Timeout waiting for the cluster to be ready for the storage migration.")
```

File: pkg/cli/cluster_migrate_multinode_storage.go [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+	}
+	readyCtx, cancel := context.WithTimeout(ctx, opts.readyTimeout)
+	defer cancel()
+	opts.log.Printf("waiting cluster to report as ready for storage migration (%s timeout).", opts.readyTimeout)
+	if err := wait.PollUntilContextCancel(readyCtx, 5*time.Second, true, readyfn); err != nil {
+		return fmt.Errorf("failed to wait for ekco to be ready for migration: %w", err)
+	}
+	opts.log.Printf("approving cluster storage migration.")
+	if err := approveStorageMigration(opts); err != nil {
+		log.Fatal(err)
```
Comment (emosbaugh) [inline pkg/cli/cluster_migrate_multinode_storage.go:203]: why fatal and not return an error here?

Comment (ricardomaraschini) [reply pkg/cli/cluster_migrate_multinode_storage.go:203]: Oh, that was a mistake.

Comment (ricardomaraschini) [reply pkg/cli/cluster_migrate_multinode_storage.go]: Fixed.

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**